pkg/dogstatsd/listeners/udp.go

```diff
@@ -61,9 +64,10 @@ func NewUDPListener(packetOut chan *Packet, packetPool *PacketPool) (*UDPListene
 	}
 
 	listener := &UDPListener{
-		packetOut:  packetOut,
-		packetPool: packetPool,
-		conn:       conn,
+		packetOut:    packetOut,
+		packetPool:   packetPool,
+		packetBuffer: newPacketBuffer(4096, 100*time.Millisecond, packetOut),
```

Comment:
I would probably want to make both the buffer size + the time-triggered flush interval configurable, maybe not documented but probably configurable. It would give us a knob to improve performance for heavy DSD users, or users with uncharacteristic DSD behavior.